apps/meteor/client/views/room/contextualBar/Info/hooks/useRoomActions.ts

```diff
@@ -8,75 +8,111 @@ import { useRoomHide } from './actions/useRoomHide';
 import { useRoomLeave } from './actions/useRoomLeave';
 import { useRoomMoveToTeam } from './actions/useRoomMoveToTeam';
 
-type RoomActions = {
+type UseRoomActionsOptions = {
 	onClickEnterRoom?: () => void;
 	onClickEdit?: () => void;
+	resetState?: () => void;
 };
 
-export const useRoomActions = (room: IRoom, { onClickEnterRoom, onClickEdit }: RoomActions, resetState?: () => void) => {
+/**
+ *
+ * @param room
+ * @param param1
+ * @param resetState
+ * @returns If more than two room actions are enabled `menu.regular` will be a non-empty array
+ */
+export const useRoomActions = (room: IRoom, { onClickEnterRoom, onClickEdit, resetState }: UseRoomActionsOptions) => {
 	const t = useTranslation();
+	const size = 2;
 
 	const handleHide = useRoomHide(room);
 	const handleLeave = useRoomLeave(room);
 	const { handleDelete, canDeleteRoom } = useDeleteRoom(room, { reload: resetState });
 	const handleMoveToTeam = useRoomMoveToTeam(room);
 	const handleConvertToTeam = useRoomConvertToTeam(room);
 
-	const memoizedActions = useMemo(
-		() => ({
-			...(onClickEnterRoom && {
-				enter: {
-					label: t('Enter'),
-					icon: 'login' as const,
-					action: onClickEnterRoom,
-				},
-			}),
-			...(onClickEdit && {
-				edit: {
-					label: t('Edit'),
-					icon: 'edit' as const,
-					action: onClickEdit,
-				},
-			}),
-			...(canDeleteRoom &&
-				handleDelete && {
-					delete: {
-						label: t('Delete'),
+	return useMemo(() => {
+		const memoizedActions = [
+			...(handleHide
+				? [
+						{
+							id: 'hide',
+							name: t('Hide'),
+							icon: 'eye-off' as const,
+							action: handleHide,
+						},
+				  ]
+				: []),
+			...(onClickEnterRoom
+				? [
+						{
+							id: 'enter',
+							name: t('Enter'),
+							icon: 'login' as const,
+							action: onClickEnterRoom,
+						},
+				  ]
+				: []),
+			...(onClickEdit
+				? [
+						{
+							id: 'edit',
+							name: t('Edit'),
+							icon: 'edit' as const,
+							action: onClickEdit,
+						},
+				  ]
+				: []),
+			...(handleLeave
+				? [
+						{
+							id: 'leave',
+							name: t('Leave'),
+							icon: 'sign-out' as const,
+							action: handleLeave,
+						},
+				  ]
+				: []),
+			...(handleMoveToTeam
+				? [
+						{
+							id: 'move_channel_team',
+							name: t('Teams_move_channel_to_team'),
+							icon: 'team-arrow-right' as const,
+							action: handleMoveToTeam,
+						},
+				  ]
+				: []),
+			...(handleConvertToTeam
+				? [
+						{
+							id: 'convert_channel_team',
+							name: t('Teams_convert_channel_to_team'),
+							icon: 'team' as const,
+							action: handleConvertToTeam,
+						},
+				  ]
+				: []),
+		];
+
+		if (memoizedActions.length <= size) {
```

Comment:
**this shouldn't live here**, the hook should just return the actions! 
This rule to display the actions should live outside here

Reply:
Why not? We are avoiding some unnecessary computations with an "Early Return" with the right object values...

Reply:
The computation here is minimum since the actions are memoized. 
Ideally the hook should do a isolated job, such as return all possible actions. 

In this case you're assuming every single time we want to use this hook we'll get the actions divided based on the very specific render condition on this view, not allowing us to render the actions differently or having to do some magic when trying another approach.

Reply:
this hook is so specific that I think there is no necessity to do a different approach... and it's possible to adjust the division passing the `size` option to the hook.

Reply:
I'm sorry, but I don't agree with it. I understand we can't keep using `useActionsSpread` since this hook was created using a different pattern, but IMO it should be handled separately. Because of that I can't approve it.

Reply:
@dougfabris That split was a parameter coming from `useActionsSpread`, `size`. You're arguing the hook should not split it or provide a way to customize the value? If the former, it's kinda pointless, since we already split the actions for the rendering logic as `regular` e `danger`; if the latter, promoting the constant as a parameter is adding an accidental complexity, since there is no reuse of `useRoomActions` anywhere else _yet_.

Reply:
> Ideally the hook should do a isolated job, such as return all possible actions.

That's why `useActionsSpread` existed, because it was carrying the rendering logic while `useRoomActions` was dealing only with building the data structure. Not sure if we gain anything splitting those again for use in a single place.

Reply:
Splitting is fine, I'm just preaching the gospel: https://overreacted.io/the-wet-codebase/